backend/packages/Upgrade/src/api/controllers/ExperimentController.ts

```diff
@@ -828,11 +829,24 @@ export class ExperimentController {
    */
   @Get('/single/:id')
   @OnUndefined(ExperimentNotFoundError)
-  public one(
+  public async one(
     @Params({ validate: true }) { id }: ExperimentIdValidator,
     @Req() request: AppRequest
   ): Promise<ExperimentDTO> {
-    return this.experimentService.getSingleExperiment(id, request.logger);
+    const experiment = await this.experimentService.getSingleExperiment(id, request.logger);
+
+    if (SUPPORTED_MOOCLET_ALGORITHMS.includes(experiment.assignmentAlgorithm)) {
+      if (!env.mooclets?.enabled) {
```

Comment:
This `if` check should be a util function which should throw an error if the experiment contains a valid mooclet algorithm and mooclet is not enabled, otherwise return true.  
It is getting in the `create` function as well and even Ben is using this in his [PR](https://github.com/CarnegieLearningWeb/UpGrade/pull/2205/files#diff-e29da6f404aa3e3fd4a253786e23cf2fda0d8e8e09d4d0ca3ecfcd5aab43c1b4)

Reply:
sure that makes sense, let me make a separate ticket for that for follow-up